hdfs/src/main/scala/akka/stream/alpakka/hdfs/model.scala

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.hdfs
+
+import java.util.function.BiFunction
+
+import akka.NotUsed
+import akka.stream.alpakka.hdfs.HdfsWritingSettings._
+import akka.stream.alpakka.hdfs.impl.strategy.DefaultRotationStrategy._
+import akka.stream.alpakka.hdfs.impl.strategy.DefaultSyncStrategy._
+import akka.stream.alpakka.hdfs.impl.strategy.Strategy
+import org.apache.hadoop.fs.Path
+
+import scala.concurrent.duration.FiniteDuration
+
+final case class HdfsWritingSettings(
+    overwrite: Boolean = true,
+    newLine: Boolean = false,
+    pathGenerator: FilePathGenerator = DefaultFilePathGenerator
+) {
+  def withOverwrite(overwrite: Boolean): HdfsWritingSettings = copy(overwrite = overwrite)
+  def withNewLine(newLine: Boolean): HdfsWritingSettings = copy(newLine = newLine)
+  def withPathGenerator(generator: FilePathGenerator): HdfsWritingSettings = copy(pathGenerator = generator)
+}
+
+object HdfsWritingSettings {
+
+  private val DefaultFilePathGenerator: FilePathGenerator =
+    FilePathGenerator((rc: Long, _: Long) => s"/tmp/alpakka/$rc")
+
+  /**
+   * Java API
+   */
+  def create(): HdfsWritingSettings =
+    HdfsWritingSettings()
+}
+
+final case class IncomingMessage[T, C](source: T, passThrough: C)
```

Comment:
I'm not super happy with the `IncomingMessage` name used in several connectors. It can easily become messy when using several connectors in the same codebase. And "incoming" is tied to the point of view of the stage, for the user the data leaves...
Maybe `HdfsWriteMessage` or `HdfsData`?